### PR TITLE
Search components with at least 3 characters

### PR DESCRIPTION
**Pull request by teryk**
WS impacted:

- api/measures/component_tree
- api/components/tree

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeAction.java [hunk 1]
```diff
@@ -243,6 +245,8 @@ private static ComponentTreeWsRequest toComponentTreeWsRequest(Request request)
       .setPageSize(request.mandatoryParamAsInt(Param.PAGE_SIZE))
       .setQuery(request.param(Param.TEXT_QUERY));
     checkRequest(componentTreeWsRequest.getPageSize() <= MAX_SIZE, "The '%s' parameter must be less than %d", Param.PAGE_SIZE, MAX_SIZE);
+    checkRequest(componentTreeWsRequest.getQuery() == null || componentTreeWsRequest.getQuery().length() >= QUERY_MINIMUM_LENGTH,
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeAction.java]: BLOCKER NullPointerException might be thrown as 'getQuery' is nullable here [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* BLOCKER 1 blocker
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'deprecated' package (org.sonarsource.sonarqube:sonar-db:src/main/java/org/sonar/db/deprecated) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* BLOCKER 1 blocker
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'deprecated' package (org.sonarsource.sonarqube:sonar-db:src/main/java/org/sonar/db/deprecated) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by julienlancelot** [comment]
@teryk Minor feedbacks for you (in fact the same as SonarTech !)

**Comment by julienlancelot** [comment]
@stas some feedbacks for you (I've added a comment in the JIRA ticket)

**Comment by julienlancelot** [comment]
Be careful, the PR and the branch jobs are red.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'deprecated' package (org.sonarsource.sonarqube:sonar-db:src/main/java/org/sonar/db/deprecated) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'deprecated' package (org.sonarsource.sonarqube:sonar-db:src/main/java/org/sonar/db/deprecated) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by julienlancelot** [comment]
LGTM

**Pull request closed without merging**